### DISCUSSION TITLE
wallet_addEthereumChain failing

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1960] [Developer Support]
(Question from Discord)

If Network Already exists in meta mask => Switch Chain Happens - No problem
If Network Doesn't Exist => Nothing Happens, If I try to do provider.request("wallet_addEthereumChain") it fails and gives an error "The Provider does not have a rpcUrl to make a request for the requested method: wallet_addEthereumChain" or sometimes there is no response or Nothing Happens.
We have observed similar behaviour across various clients including wagmi, web3-onboard, web-react and even web3modal, which feels such a simple use-case but doesn't works

**Comment by saisumit** [comment]
Can you guys prioritise this, we are facing urgent issues on our production environment.
